### DISCUSSION TITLE
Automatically create and assign licenses

### DISCUSSION THREAD

**Discussion by r-brown** [Labs64/NetLicensing-Community#93] [Questions] [answered]
**Question**

> If have to add 50 more licenses of n number of users, then it would be difficult for them to add a large amount no. of licenses manually without using the try and buy module because in this module I have to add an individual module or every user so this is also difficult.
> so is there any provision to create a license automatically

**Comment by r-brown** [comment, marked as answer]
NetLicensing provides [RESTful API](https://netlicensing.io/wiki/restful-api) which can be used to automate your license management processes.

In order to create Licenses in a bulk we advice you to use [LicenseService.create()](https://netlicensing.io/wiki/license-services#create-license) method.

**Comment by Aakansha43** [comment]
in subscription model when my time volume period is for 1 year then after validating  my licensee why it is not shown after one-year expiration date

**Comment by r-brown** [comment]
All licenses are available in the NetLicensing Management Console.
However, expired licenses will be shown as inactive.

Validation response will return consolidated validation result containing total expiration date - not single licenses.
